libcloud/compute/drivers/cloudstack.py

```diff
@@ -3489,70 +3489,108 @@ def ex_delete_security_group(self, name):
                                   params={'name': name},
                                   method='GET')['success']
 
-    def ex_authorize_security_group_ingress(self, securitygroupname,
-                                            protocol, cidrlist, startport,
-                                            endport=None):
+    def ex_authorize_security_group_ingress(self, securitygroupname, protocol,
+                                            cidrlist, startport=None,
+                                            endport=None, icmptype=None,
+                                            icmpcode=None, **kwargs):
```

Comment:
Can you please explicitly declare all the supported arguments instead of using **kwargs?